codalab/model/bundle_model.py

```diff
@@ -861,17 +867,23 @@ def transition_bundle_staged(self, bundle):
             ).fetchone()
             if not row:
                 raise IntegrityError('Missing bundle with UUID %s' % bundle.uuid)
-            if row.state != State.STARTING:
-                # It is possible that this method is called on a bundle
-                # that has started running.
-                return False
 
-            update_message = {'state': State.STAGED, 'metadata': {'job_handle': None}}
-            self.update_bundle(bundle, update_message, connection)
+            # Reset all metadata fields to be None
```

Comment:
Can we programmatically get all the fields (where `generated=True`) from `run_bundle.py` and clear them?

Reply:
There is one issue of deleting all the fields where `generated=True`. The field `action` is marked with `generated=True`, but it's displayed in the `staged` bundle page.

Reply:
I think it's fine to clear it, since `action` is really specific to the particular execution of a bundle (e.g., kill).  And even if we wanted to preserve it, we could still programmatically get the fields ad just exclude that one.

Reply:
Yeah, that's true. I updated this part to get all the fields with `generated=True` and exclude the `action` field for now. I guess keep the `action` field for `staged` bundle could be useful when it gets killed?